### PR TITLE
*: Fix bug in show create table

### PR DESCRIPTION
**Pull request by shenli**
Fix https://github.com/pingcap/tidb/issues/1918
@coocood @tiancaiamao @zimulala PTAL

### CODE REVIEW

**Comment by hanfei1991** [comment]
LGTM

File: executor/show.go [hunk 1]
```diff
@@ -388,7 +388,9 @@ func (e *ShowExec) fetchShowCreateTable() error {
 			if !mysql.HasNoDefaultValueFlag(col.Flag) {
 				switch col.DefaultValue {
 				case nil:
-					buf.WriteString(" DEFAULT NULL")
+					if !mysql.HasPriKeyFlag(col.Flag) {
```
Comment (zimulala) [inline executor/show.go]: Here nesting level too deep. Can you reduce it?

Comment (hanfei1991) [inline executor/show.go]: Do you need to check !mysql.HasPriKeyFlag && !mysqlHasNotNullFlag ?

File: executor/show_test.go [hunk 2]
```diff
@@ -44,6 +44,24 @@ func (s *testSuite) TestShow(c *C) {
...
+		c varchar(10) NOT NULL, 
+		d time unique
+	);`
+	tk.MustExec(testSQL)
+	testSQL = "show create table ptest;"
+	result = tk.MustQuery(testSQL)
+	c.Check(result.Rows(), HasLen, 1)
+	row = result.Rows()[0]
+	expectedRow = []interface{}{
+		"ptest", "CREATE TABLE `ptest` (\n  `a` int(11) NOT NULL,\n  `b` double NOT NULL DEFAULT '2',\n  `c` varchar(10) NOT NULL,\n  `d` time DEFAULT NULL,\n PRIMARY KEY (`a`),\n  UNIQUE KEY `d` (`d`)\n) ENGINE=InnoDB"}
```
Comment (zimulala) [inline executor/show_test.go]: Does it need to add charset information?

File: executor/show.go (hunk 1 above)
Comment (coocood) [inline executor/show.go]: How about check `!mysql.HasNotNullFlag(col.Flag)` instead?

Comment (tiancaiamao) [inline executor/show.go]: primary key column can't be NULL...
if a column has `mysql.HasPriKeyFlag`, it must have `mysql.HasNotNullFlag`

Comment (coocood) [inline executor/show.go]: May be the check `!mysql.HasNoDefaultValueFlag(col.Flag)` can be removed.

**Comment by tiancaiamao** [comment]
LGTM

**Comment by coocood** [comment]
LGTM

**Comment by zimulala** [comment]
LGTM

**Pull request merged**